contributors/design-proposals/daemonset-update.md

```diff
@@ -129,18 +127,35 @@ type DaemonSetStatus struct {
 
 	// UpdatedNumberScheduled is the total number of nodes that are running updated
 	// daemon pod
+	// +optional
 	UpdatedNumberScheduled int32 `json:"updatedNumberScheduled"`
 
 	// NumberAvailable is the number of nodes that should be running the
 	// daemon pod and have one or more of the daemon pod running and
 	// available (ready for at least minReadySeconds)
+	// +optional
 	NumberAvailable int32 `json:"numberAvailable"`
 
 	// NumberUnavailable is the number of nodes that should be running the
 	// daemon pod and have non of the daemon pod running and available
 	// (ready for at least minReadySeconds)
+	// +optional
 	NumberUnavailable int32 `json:"numberUnavailable"`
 }
+
+type DaemonSet struct {
```

Comment:
> Could we consider putting the field under the DaemonSetStatus?

Not sure how to do that in API server. [Update strategy](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/extensions/daemonset/strategy.go#L58-L80) forbids updating status, and [status update strategy](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/extensions/daemonset/strategy.go#L141-L145) ignores changes not in status. Also, we update DaemonSet status in DaemonSet controller sync loop, not in API server. 

Maybe we could update templateGeneration in status when we get notification from informer that DaemonSet template is updated, assuming informer is reliable and never misses any event?